pandapipes/test/topology/test_nxgraph.py

```diff
@@ -19,8 +20,18 @@ def test_include_branches():
 
     mg = top.create_nxgraph(net, include_pipes=True, include_valves=True, include_pumps=False)
     assert len(mg.nodes()) == len(net.junction)
-    assert len(mg.edges()) == len(net.pipe) + len(net.valve) 
+    assert len(mg.edges()) == len(net.pipe)
+
+    mg = top.create_nxgraph(net, include_pipes=True, include_valves=True, include_pumps=False,
+                            neglect_in_service=["valve"])
```

Comment:
Instead of neglect_in_service, I would rather do it like in plotting: respect_in_service. That would mean for me, if I set include_valves=True, then I would say all edges are considered, no matter if they are in_service or not. However, when to say, respect_in_servce=['valve'], I would  only take the valves which are in_service. As in this particular example, both valves are in_service, therefore, I would say, in both cases len(mg.edges()) == len(net.pipe) + len(net.valve) should be correct. If you set both out-of-service and set respect_in_service=['valves'], assert len(mg.edges()) == len(net.pipe)  should be correct.

Reply:
I don't really agree with that point for two reasons:
1. It would alter the behavior from pandapower, making it harder for users of pandapower to use the functionality. In pandapower, in_service is always respected and branches are excluded in that case.
2. To me, the pandapower beahvior is more straight-forward. However, in pandapower, neglecting the in_service flag is only possible for a hand full of elements and it's hard coded.
Maybe we can combine it with your next comment and somehow add keyword arguments that can respect the behavior for certain components...

Reply:
Okay, I see. I actually don't really like, that plotting and topology is the exact opposite. In plotting you can set respect_in_service, because the default option is that everything is considered. In topology, however, default is to only consider in_service and then neglect only in_service if the flag is set. I think, we should be more consistent. However, if it is done like this in pandapower, then we should stick to that. Might be a aspect that can be considered in pandapower then.